lib/handler.ex

```diff
@@ -1,4 +1,6 @@
 defmodule HTTPServer.Handler do
   alias HTTPServer.Request
-  @callback handle(req :: %Request{}) :: {status_code :: integer(), body :: term()}
+
+  @callback handle(req :: %Request{}) ::
+              {status_code :: integer(), body :: term(), headers :: map()}
```

Comment:
I like this move (at the moment). With the current design, the headers for different endpoints and methods will be different, and the user would want control.